### PR TITLE
adding Find-BBServerRepository and tests

### PR DESCRIPTION
**Pull request by scott-pickthorn**
- adding Find-BBServerRepository that returns all the repositorys that match a repositoryName or wildcard.

### CODE REVIEW

File: BitbucketServerAutomation/BitbucketServerAutomation.psd1 [hunk 1]
```diff
@@ -12,25 +12,25 @@
```
Comment (splatteredbits) [inline BitbucketServerAutomation/BitbucketServerAutomation.psd1:12]: Please undo all the whitespace changes. What caused this?

File: BitbucketServerAutomation/Functions/Find-BBServerRepository.ps1 [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        [Object]
+        $Connection,
+        # Name of a Repository you wish to find. supports Wildcards. 
+        [String]
+        $Name
+    )
+    $repositoryList = @()
+    Get-BBServerProject -Connection $Connection |
+        ForEach-Object {
+        $repositoryList += Get-BBServerRepository -Connection $Connection -ProjectKey $_.key| Where-Object { $_.name -like $Name }
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Find-BBServerRepository.ps1]: Indent this line and the next.

File: BitbucketServerAutomation/Functions/Find-BBServerRepository.ps1 [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+{
+    param(
+        [parameter(Mandatory=$true)]
+        [Object]
+        $Connection,
+        # Name of a Repository you wish to find. supports Wildcards. 
+        [String]
+        $Name
+    )
+    $repositoryList = @()
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Find-BBServerRepository.ps1]: Remove `$repositoryList`. Let the pipeline return any repository object that matches the `Where-Object` condition.

File: Tests/Find-BBServerRepository.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    GivenARepository -Name 'first'
+    WhenARepositoryIsRequested -RequestedRepo 'badRepo'
+    ThenNoRepositoryShouldBeReturned
+}
+
+Describe 'Find-BBServerRepository.When searching for a repository that matches a single repository' {
+    GivenARepository -Name 'first'
+    GivenARepository -Name 'second'
+    WhenARepositoryIsRequested -RequestedRepo 'second'
+    ThenTheRepositoryShouldBeLike -name 'second'
```
Comment (splatteredbits) [inline Tests/Find-BBServerRepository.Tests.ps1]: Change this to something like `ThenRepositoryReturned`. You pass it a list of names of repositories that should be returned. No wildcard matching required/needed.

File: BitbucketServerAutomation/Functions/Find-BBServerRepository.ps1 [hunk 5]
```diff
@@ -0,0 +1,17 @@
+function Find-BBServerRepository
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Find-BBServerRepository.ps1:13]: Missing license notice.

File: Tests/Find-BBServerRepository.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,84 @@
+& (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-BitbucketServerAutomationTest.ps1' -Resolve)
```
Comment (splatteredbits) [inline Tests/Find-BBServerRepository.Tests.ps1:13]: Missing license notice.

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: BitbucketServerAutomation/Functions/Find-BBServerRepository.ps1 (hunk 5 above)
Comment (scott-pickthorn) [reply BitbucketServerAutomation/Functions/Find-BBServerRepository.ps1:13]: Done

File: BitbucketServerAutomation/BitbucketServerAutomation.psd1 (hunk 1 above)
Comment (scott-pickthorn) [reply BitbucketServerAutomation/BitbucketServerAutomation.psd1:12]: Done

File: Tests/Find-BBServerRepository.Tests.ps1 (hunk 6 above)
Comment (scott-pickthorn) [reply Tests/Find-BBServerRepository.Tests.ps1:13]: Done

**Review by splatteredbits — APPROVED** [review]

**Pull request merged**